sky/skylet/providers/ibm/node_provider.py

```diff
@@ -0,0 +1,924 @@
+        """
+
+        # collecting valid nodes that are either starting, running or pending.
+        res_nodes = []
+
+        found_nodes = self._get_nodes_by_tags(tag_filters)
+
+        for node in found_nodes:
+
+            # check if node scheduled for delete
+            with self.lock:
+                if node["id"] in self.deleted_nodes:
+                    logger.info(f"{node['id']} scheduled for delete")
+                    continue
+
+            # validate instance in correct state
+            valid_statuses = ["pending", "starting", "running"]
+            if node["status"] not in valid_statuses:
+                with self.lock:
+                    if node["status"] == "failed":
+                        # since non_terminated_nodes is called periodically,
+                        # prevent access until lock until node is deleted
+                        self._delete_node(node["id"])
+                        logger.error(
+                            """Encountered a failed node.
+                            Region might be crowded,
+                            Consider retrying later.""")
+                    else:
+                        logger.debug(
+                            f"{node['id']} status {node['status']}"
+                            f" not in {valid_statuses}, skipping"
+                        )
+                        continue
+
+            # validate instance not hanging in pending state
+            with self.lock:
+                if node["id"] in self.pending_nodes:
+                    if node["status"] != "running":
+                        pending_time = time.time() - \
+                            self.pending_nodes[node["id"]]
+                        logger.debug(
+                            f"{node['id']} is pending for {pending_time}")
+                        if pending_time > PENDING_TIMEOUT:
+                            logger.error(
+                                f"pending timeout {PENDING_TIMEOUT} reached, "
+                                f"deleting instance {node['id']}"
+                            )
+                            # we won't try to restart a failed node even if
+                            self._delete_node(node["id"])
+                            # avoid adding the node to cached_nodes
+                            continue
+                    else:
+                        self.pending_nodes.pop(node["id"], None)
+
+            # skypilot-adjustment: verify ip is attached to worker nodes as well
+            nic_id = node["network_interfaces"][0]["id"]
+
+            # pylint: disable=line-too-long
+            res = self.ibm_vpc_client.list_instance_network_interface_floating_ips(
+                node["id"], nic_id
+            ).get_result()
+
+            floating_ips = res["floating_ips"]
+            if len(floating_ips) == 0:
+                # not adding a node that's missing floating ip
```

Comment:
it means provisioning process have yet to finish. 
since `non_terminated_nodes` is called periodically, the node will be added to the return list next time around.